lib/gym/build_command_generator.rb

```diff
@@ -76,8 +76,14 @@ def build_path
       end
 
       def archive_path
+        config = Gym.config
+        unless @archive_path
+          if config[:archive_path]
+            @archive_path = config[:archive_path]
+          end
+        end
         unless @archive_path
-          file_name = [Gym.config[:output_name], Time.now.strftime("%F %H.%M.%S")] # e.g. 2015-08-07 14.49.12
+          file_name = [config[:output_name], Time.now.strftime("%F %H.%M.%S")] # e.g. 2015-08-07 14.49.12
           @archive_path = File.join(build_path, file_name.join(" ") + ".xcarchive")
```

Comment:
You described the parameter to be the directory in which the archive should be store. But as it seems you don't append `.xcarchive` to the path?